source/agora/test/BanManager.d

```diff
@@ -79,42 +80,42 @@ unittest
     // leftover txs which node 3 will reject due to its filter
     Transaction[] left_txs;
 
-    auto new_tx = genBlockTransactions(4);
-    left_txs ~= new_tx;
-    new_tx.each!(tx => node_1.putTransaction(tx));
+    foreach (block_idx; 0 .. 4)
+    {
+        auto new_tx = genBlockTransactions(1);
+        left_txs ~= new_tx;
+        new_tx.each!(tx => node_1.putTransaction(tx));
+
+        [node_1, node_2].enumerate.each!((idx, node) =>
+            retryFor(node.getBlockHeight() == 1 + (block_idx + 1),
+                4.seconds,
+                format("Block %s Node %s has block height %s. Expected: %s",
+                    block_idx + 1, idx, node.getBlockHeight().to!string, 1 + (block_idx + 1))));
+
+        retryFor(node_3.getBlockHeight() == 1, 1.seconds, node_3.getBlockHeight().to!string);
+    }
 
     // wait for node 3 to be banned and all putTransaction requests to time-out
     Thread.sleep(2.seconds);
 
-    import std.conv;
-    retryFor(node_1.getBlockHeight() == 5, 1.seconds, node_1.getBlockHeight().to!string);
-    retryFor(node_2.getBlockHeight() == 5, 1.seconds, node_2.getBlockHeight().to!string);
-    retryFor(node_3.getBlockHeight() == 1, 1.seconds, node_3.getBlockHeight().to!string);
-
-    // clear putTransaction filter
+    // sanity check: block height should not updated, node 3 is not a validator and cannot make blocks
```

Comment:
Maybe add your explanation that the Ledger can only be manipulated through `getBlockFrom` ?